### ISSUE TITLE
strange scanning behaviour with long integration time

### ISSUE THREAD

**Issue by pjb671** [sardana-org/sardana#1125]
Hi,

I have encountered a strange behaviour testing 2.7.2-alpha here at maxiv, if I try to use integration time > 10s in a step scan. I noticed it first with the real detectors but can easily reproduce it with the standard dummy controllers. I have a measurement group with the uxtimer and a dummy 2d element.  If I scan a dummy motor for a few steps with an integration time less than 10s then all is OK:

Door_nanolab_1 [200]: %ascan dummy_motor 1 10 3 1
Scan #145 started at Mon Jun 3 11:44:13 2019. It will take at least 0:00:05.269742
#Pt No dummy_motor uxtimer dummy2d dt
0 1 1 (1024, 1024) 0.479495
1 4 1 (1024, 1024) 2.02885
2 7 1 (1024, 1024) 3.61212
3 10 1 (1024, 1024) 5.13796
Scan #145 ended at Mon Jun 3 11:44:19 2019, taking 0:00:06.382351. Dead time 37.3% (motion dead time 20.7%)

Door_nanolab_1 [201]: %ascan dummy_motor 1 10 3 10
Scan #146 started at Mon Jun 3 11:44:21 2019. It will take at least 0:00:41.269742
#Pt No dummy_motor uxtimer dummy2d dt
0 1 9.96915 (1024, 1024) 0.477726
1 4 9.93483 (1024, 1024) 10.7921
2 7 9.93586 (1024, 1024) 21.0973
3 10 9.91786 (1024, 1024) 31.4195
Scan #146 ended at Mon Jun 3 11:45:03 2019, taking 0:00:41.441601. Dead time 3.5% (motion dead time 3.2%)

However, if I try with 20s then the stepping gets out of sync with the detector reading: 

Door_nanolab_1 [202]: %ascan dummy_motor 1 10 3 20
Scan #147 started at Mon Jun 3 11:45:06 2019. It will take at least 0:01:21.269742
#Pt No dummy_motor uxtimer dummy2d dt
0 1 9.97506 (1024, 1024) 0.483076
1 4 9.9218 (1024, 1024) **10.783**
2 7 9.9185 (1024, 1024) 31.0245
3 10 9.91543 (1024, 1024) 51.2392
Scan #147 ended at Mon Jun 3 11:46:17 2019, taking 0:01:11.184054. Dead time -12.4% (motion dead time 1.8%)

In this case, if I run the pool in a terminal so I can see what's happening, the steps advance 10s before the ReadOne is called. It means that in the end, after Scan #147 ended is printed in spock, the pool remains busy for 10s after which the final ReadOne is called. 

This is pretty strange. I have been using 2.5.0 until now and have never seen anything like this.
cheers,
Paul

**Comment by pjb671** [comment]
...and here is the expected behaviour when I rolled back to 2.5.0:

Door_nanolab_1 [4]: %ascan dummy_motor 1 10 3 20
Scan #150 started at Mon Jun 3 14:50:22 2019. It will take at least 0:01:21.269742
#Pt No dummy_motor uxtimer dummy2d dt
0 1 20 (1024, 1024) 0.475682
1 4 20 (1024, 1024) **21.0854**
2 7 20 (1024, 1024) 41.7439
3 10 20 (1024, 1024) 62.2867
Scan #150 ended at Mon Jun 3 14:51:45 2019, taking 0:01:22.567755. Dead time 3.1% (motion dead time 1.6%)

**Comment by reszelaz** [comment]
Thanks for reporting this @pjb671 !

I can not reproduce it with the most recent development branch:

```
Door_test_1 [1]: ascan mot09 1 10 3 20
ScanDir is not defined. This operation will not be stored persistently. Use Use "expconf" (or "senv ScanDir <abs directory>") to enable it
Scan #2 started at Mon Jun  3 21:37:18 2019. It will take at least 0:01:21.469694
 #Pt No    mot09      ct09     twod03      dt   
   0         1         20     (1024, 1024)  0.0150928
   1         4         20     (1024, 1024)  20.7294 
   2         7         20     (1024, 1024)  41.4327 
   3         10        20     (1024, 1024)  62.1261 
Scan #2 ended at Mon Jun  3 21:38:40 2019, taking 0:01:22.314881. Dead time 2.8% (motion dead time 1.8%)

Door_test_1 [2]: ascan mot09 1 10 3 10
ScanDir is not defined. This operation will not be stored persistently. Use Use "expconf" (or "senv ScanDir <abs directory>") to enable it
Scan #3 started at Mon Jun  3 21:38:45 2019. It will take at least 0:00:42.318222
 #Pt No    mot09      ct09     twod03      dt   
   0         1         10     (1024, 1024)  0.870795
   1         4         10     (1024, 1024)  11.5666 
   2         7         10     (1024, 1024)  22.2611 
   3         10        10     (1024, 1024)  32.9528 
Scan #3 ended at Mon Jun  3 21:39:28 2019, taking 0:00:43.150710. Dead time 7.3% (motion dead time 5.5%)
```
Could you try this?

Also the `2.7.2-alpha` version does not say too much. We use the alpha version number for the whole period in between the official releases. So I would need a commit hash to be able to reproduce it. But if it is fixed in the latest develop, please confirm that, I would suggest to simply update.

**Comment by aureocarneiro** [comment]
Hey @reszelaz the commit hash that we used to build this version that Paul tested (2.7.2-alpha) was: 9316dc188b66833d049a9220e8924454ef92ea79 , I will try to reproduce the error today, thanks!

**Comment by reszelaz** [comment]
I've just checked with the 9316dc1 and I do not observe this behavior.

```
Door_zreszela_1 [2]: defmeas mntgrp30 ct01 twod01
Created mntgrp30

Door_zreszela_1 [3]: senv ActiveMntGrp mntgrp30
ActiveMntGrp = mntgrp30

[...]

Door_zreszela_1 [14]: ascan mot01 1 10 3 20
Operation will be saved in /tmp/test_midtrigger.h5 (HDF5::NXscan from NXscanH5_FileRecorder)
Scan #7 started at Tue Jun 11 15:11:17 2019. It will take at least 0:01:38.400000
 #Pt No    mot01      ct01     twod01      dt   
   0         1         20     (1024, 1024)  9.13629 
   1         4         20     (1024, 1024)  32.6184 
   2         7         20     (1024, 1024)   56.106 
   3         10        20     (1024, 1024)  79.6155 
Operation saved in /tmp/test_midtrigger.h5 (HDF5::NXscan)
Scan #7 ended at Tue Jun 11 15:12:57 2019, taking 0:01:39.990864. Dead time 20.0% (motion dead time 18.5%)

Door_zreszela_1 [15]: ascan mot01 1 10 3 10
Operation will be saved in /tmp/test_midtrigger.h5 (HDF5::NXscan from NXscanH5_FileRecorder)
Scan #8 started at Tue Jun 11 15:14:25 2019. It will take at least 0:00:58.400000
 #Pt No    mot01      ct01     twod01      dt   
   0         1         10     (1024, 1024)  9.14605 
   1         4         10     (1024, 1024)  22.6854 
   2         7         10     (1024, 1024)  36.1796 
   3         10        10     (1024, 1024)  49.6666 
Operation saved in /tmp/test_midtrigger.h5 (HDF5::NXscan)
Scan #8 ended at Tue Jun 11 15:15:25 2019, taking 0:01:00.082365. Dead time 33.4% (motion dead time 30.8%)
```
How do you create the measurement group?
Maybe you have Tango polling enabled on states of the channes? measurement groups?

**Comment by aureocarneiro** [comment]
Hej @reszelaz I tested with sardana from develop branch and 9316dc1 , and I can not reproduce the error.
My logs:

```
Door_nanolab_1 [1]: %ascan dummy_motor 1 10 3 20
Operation will be saved in /tmp/test.h5 (HDF5::NXscan)
Scan #185 started at Wed Jun 12 16:05:57 2019. It will take at least 0:01:20.959904
 #Pt No   dummy_motor  uxtimer   dummy2d      dt
   0           1          20     (1024, 1024)  0.179159
   1           4          20     (1024, 1024)  21.0583
   2           7          20     (1024, 1024)  41.8708
```

@pjb671 Can you give the feedback to us?

Thanks!

**Comment by pjb671** [comment]
Hi,
Thanks both for checking this. It was very strange as I saw it both with real and dummy exp channels. As Aureo says we cannot reproduce it today.  Also I noticed now that in the timer column in the example above, instead of 20 I saw numbers around 10, like 9.97506.  The only theory we have is that I was using different versions of Sardana for the server and client (ie the machine with spock), though only by one digit in the minor version. Today with the develop branch for client and master we do not see the error, and nor can we see it going back to 9316dc1. So we can mark as unreproducible and close this, and assume it must have been something wrong in my setup at the time.
cheers,
Paul
